### PR TITLE
Feature SALES-4: Implements aggregate getter endpoints for book detail

### LINKED ISSUE #20: Feature SALES-4: Implement Aggregate Getter Endpoints (2.2.2, 3.2)
2.1: Total sales to date [for a Book].

2.2.2: Totals should be shown for publisher revenue, unpaid author royalty, paid author royalty, and total author royalty.

### LINKED ISSUE #49: [Book - 6] - Move Book Sales List total calculations to the API
Totals should be shown for publisher revenue, unpaid author royalty, paid author royalty, and total author royalty.

Currently totals are calcauated on frontend, should transition to using backend as truth of information

### LINKED ISSUE #67: BOOK-8: Refactor BookDetail DTO
Design doc: https://docs.google.com/document/d/1-8hoMBtjEpugs7_H6FS1NZN4yeM8XDmIdP9V4jlwX4g/edit?tab=t.0

### PR DESCRIPTION
**Pull request by Danny1021**
Implements the aggregate financials requested by requirement 2.2.2:

"Totals should be shown for publisher revenue, unpaid author royalty, paid author royalty, and total author royalty."

These totals are calculated using SQL coalesces in the JPA repository that already existed. This PR primarily creates a record for the Java side and a DTO to transfer that information over to the frontend. The request is just the BookId you wish to aggregate the financials for and the response is the bookId followed by the values.

Includes testing.

Additionally, I took the liberty to rearrange the `RestController` classes (`SaleController.java`, `BookController.java`) as they are beginning to get a lot of endpoints.

Resolves #20 

NOTE 2/4:
Implemented design discussed in [this doc](https://docs.google.com/document/d/1-8hoMBtjEpugs7_H6FS1NZN4yeM8XDmIdP9V4jlwX4g/edit?tab=t.0).

Updated the frontend as well to remove the bloat from the front-end calculation of the values to be displayed in the financial summary and use the values derived from the `BookDetailResponse` DTO.

Kept `BookResponse` the same so as to edit the frontend code as little as possible. Now, `BookResponse` DTO refers to the Book information seen in a row of the Book List View, while `BookDetailResponse` is DTO for the Book Detail view page.

Resolves #67 

Resolves #49 

This PR also sunsets the `PagedBookResponse` object and replaces it with the generic `PagedResponse<BookResponse>` object.

### CODE REVIEW

File: backend/src/main/java/edu/duke/bookpublishing/books/BookController.java [hunk 1]
```diff
@@ -80,6 +85,17 @@ public BookResponse getBook(@PathVariable Long id) {
     return BookResponse.from(book);
   }
 
+  // BOOK DETAIL FINANCIAL AGGREGATOR
+  @Operation(
+      operationId = "getBookDetailFinancials",
```
Comment (michaelscutari) [inline backend/src/main/java/edu/duke/bookpublishing/books/BookController.java]: Let's decide on a consistent naming scheme for this feature. Maybe `BookDetailFinancials` is a good compromise. This way everything becomes, for example, `BookDetailFinancialsResponse" etc...

**Review by michaelscutari — CHANGES_REQUESTED** [review]
Great! please just rename everything to be consistent. `BookDetailFinancials` seems fine, but go for whatever you want.

File: backend/src/main/java/edu/duke/bookpublishing/books/BookController.java [hunk 2]
```diff
@@ -77,9 +93,27 @@ public BookResponse getBook(@PathVariable Long id) {
...
             .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Book not found"));
-    return BookResponse.from(book);
+
+    return BookResponse.from(book, saleService.getBookTotalSales(id));
+  }
+
+  @Operation(
+      operationId = "getBookDetailById",
+      summary = "Gets book detail information (book information + financials)")
+  @GetMapping("/bookdetail/{id}")
```
Comment (michaelscutari) [inline backend/src/main/java/edu/duke/bookpublishing/books/BookController.java]: Consider removing the `/bookdetail` mapping, and just having `/book/{id}` return a `BookDetailResponse`. Simpler API surface, plus I think the only use case of `/book/{id}` is in the detailed view anyways.

File: backend/src/main/java/edu/duke/bookpublishing/books/dto/BookDetailResponse.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @Schema(description = "ISBN-13 identifier") String isbn13,
+    @Schema(description = "ISBN-10 identifier") String isbn10,
+    @Schema(description = "Publication year") Integer publicationYear,
+    @Schema(description = "Publication month (1-12)") Integer publicationMonth,
+    @Schema(description = "Author royalty rate") BigDecimal royaltyRate,
+    @Schema(description = "Total sales quantity to date") Long totalSalesToDate,
+    @Schema(description = "Total publisher revenue earned from this book") BigDecimal revenue,
+    @Schema(description = "Total author unpaid royalty from the book") BigDecimal unpaidRoyalty,
+    @Schema(description = "Total author paid royalty from the book") BigDecimal paidRoyalty,
+    @Schema(description = "Total royalty earned by the author (both paid and unpaid")
```
Comment (michaelscutari) [inline backend/src/main/java/edu/duke/bookpublishing/books/dto/BookDetailResponse.java]: Small typo. Missing ')'

File: backend/src/test/java/edu/duke/bookpublishing/books/BookControllerTest.java [hunk 4]
```diff
@@ -9,6 +9,8 @@
 import edu.duke.bookpublishing.auth.UserRepository;
 import edu.duke.bookpublishing.books.dto.BookRequest;
```
Comment (michaelscutari) [inline backend/src/test/java/edu/duke/bookpublishing/books/BookControllerTest.java:10]: The only test that I think is missing is testing for a book that has zero sales, making sure it returns a sensible value.

File: frontend/src/pages/dashboard/crud-dashboard/data/books.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
   type BookRequest,
+  type BookDetailResponse,
   type BookResponse,
-  type PagedBookResponse,
+  type PagedResponseBookResponse,
 } from '../../../../api';
 import type { GridFilterModel, GridPaginationModel, GridSortModel } from '@mui/x-data-grid';
 
 export type Book = BookResponse;
+export type BookDetail = BookDetailResponse;
```
Comment (michaelscutari) [inline frontend/src/pages/dashboard/crud-dashboard/data/books.ts:11]: If we change to just one endpoint, then `GetDetail` can be removed/consolidated.
